civis/ml/_model.py

```diff
@@ -21,32 +22,22 @@
 except ImportError:
     HAS_SKLEARN = False
 
-from civis import APIClient, find, find_one, __version__
+from civis import APIClient, find, find_one
 from civis._utils import camel_to_snake
 from civis.base import CivisAPIError, CivisJobFailure
 from civis.compat import FileNotFoundError, TemporaryDirectory
 import civis.io as cio
 from civis.futures import ContainerFuture
+from civis.response import Response
+
 
 __all__ = ['ModelFuture', 'ModelError', 'ModelPipeline']
 log = logging.getLogger(__name__)
 
 # sentinel value for default primary key value
-SENTINEL = namedtuple('Sentinel', [])()
-
-# Map training template to prediction template so that we
-# always use a compatible version for predictions.
-_PRED_TEMPLATES = {11219: 11220,  # v2.2
-                   11221: 11220,  # v2.2 registration
-                   10582: 10583,  # v2.1
-                   9968: 9969,    # v2.0
-                   9112: 9113,    # v1.1
-                   8387: 8388,    # v1.0
-                   7020: 7021,    # v0.5
-                   }
-_CIVISML_TEMPLATE = None  # CivisML training template to use
-REGISTRATION_TEMPLATES = [11221,  # v2.2
-                          ]
+SENTINEL = collections.namedtuple('Sentinel', [])()
+
+_TEMPLATE_IDS = None  # To be updated by _get_template_ids_all_versions
```

Comment:
I don't want to trigger API calls when the module is loaded, and hence this approach. The drawback is that the following pattern is found here and there:

```python
    global _TEMPLATE_IDS
    if _TEMPLATE_IDS is None:
        _TEMPLATE_IDS = _get_template_ids_all_versions(client)
```

"Drawback" might not be the right word, because it's highly readable and maintainable to me? Feel free to let me know if there's a better way to populate `_TEMPLATE_IDS`.